pr_watch/models.py

```diff
@@ -46,22 +46,79 @@
 
 # Models ######################################################################
 
+class WatchedFork(models.Model):
+    """
+    Represents a fork of edx/edx-platform whose PRs we watch.
+    """
+    # uses internal id key
+    enabled = models.BooleanField(default=True)
+    # This is the old .env variable WATCH_ORGANIZATION
+    organization = models.CharField(
+        max_length=200,
+        db_index=True,
+        help_text='Github organization whose repositories can be watched for PRs. E.g.: open-craft'
+    )
+    # This is the old .env variable WATCH_FORK
+    fork = models.CharField(
+        max_length=200,
+        db_index=True,
+        help_text='Github fork name that will be watched for PRs. E.g.: open-craft/edx-platform'
+    )
+    # This is equivalent to the DEFAULT_CONFIGURATION_REPO_URL .env variable
+    configuration_source_repo_url = models.URLField(
+        blank=True,
+        null=True,
+        help_text='If set, it overrides the default configuration repository',
+    )
+    # Equivalent to settings.DEFAULT_OPENEDX_RELEASE
+    configuration_version = models.CharField(
+        max_length=200,
+        blank=True,
+        null=True,
+        help_text='If set, it overrides the default configuration version',
+    )
+    configuration_extra_settings = models.TextField(
+        blank=True,
+        null=False,
+        default="",
+        help_text="YAML string with extra variables. These take precedence over the others",
+    )
+
+    class Meta:
+        unique_together = ('organization', 'fork')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.logger = ModelLoggerAdapter(logger, {'obj': self})
+
+    def __str__(self):
+        return "Fork {} ({})".format(self.fork, self.organization)
+
+
 class WatchedPullRequestQuerySet(models.QuerySet):
     """
     Additional methods for WatchedPullRequest querysets
     Also used as the standard manager for the WatchedPullRequest model
     """
-    def get_or_create_from_pr(self, pr):
+    def get_or_create_from_pr(self, pr, watched_fork):
         """
         Get or create an instance for the given pull request
         """
         watched_pr, created = self.get_or_create(
             fork_name=pr.fork_name,
             branch_name=pr.branch_name,
             github_pr_url=pr.github_pr_url,
+            watched_fork=watched_fork,
         )
         if created:
             watched_pr.update_instance_from_pr(pr)
```

Comment:
Ok, I know the JIRA said to put these lines *after* this line, but I misspoke when I approved that detail.  My apologies!

I think the PR settings should take precedence over everything else, because the PR author knows what needs to be set specially for each PR.  E.g. if I have an `edx-platform` PR that depends on another change I'm submitting to `edx/configuration`, then I want my sandbox instance to use my `open-craft/configuration` branch, not the default `watched_fork.configuration_version` branch.

I'll note the changes required to the tests below too.